plugin.video.vtm.go/resources/lib/addon.py

```diff
@@ -19,13 +20,39 @@
 @routing.route('/')
 def index():
     """ Show the profile selection, or go to the main menu. """
-    if not kodiutils.has_credentials() or (kodiutils.get_setting_bool('auto_login') and bool(kodiutils.get_setting('profile'))):
-        # If we have no credentials (browse only mode) or we have autologin and a profile, go directly to the main menu
-        show_main_menu()
+    while True:
```

Comment:
this has the chance of locking kodi completely isn't it? What if someone closes kodi while in the loop? I think you need to check for abortRequest here

Reply:
We are asking the user for confirmation, and if he doesn't want to enter the credentials, we `return` out of the loop. The loop is just to ask the user to update his credentials again when they are incorrect.

I've tried to stop kodi, and it terminates just fine (without a timeout). I think Kodi is handling this since we are waiting for user input with the `yesno_dialog`.

Reply:
I tend to agree with @enen92 on his comment here. I'm not sure how to test to confirm this though.

Reply:
I've installed the addon without credentials and opened it. It popped up the question to fill in the details. Pressing esc, back or "No" stops the loop and returns to the Homescreen. I've also tried to send a SIGTERM signal to Kodi while the dialog was shown, and it gracefully stopped Kodi.

I don't see an issue here.

Reply:
Nevertheless and to avoid black magic can you loop over abortrequested instead?

Reply:
@enen92 I understand what you mean, but there isn't anything magic happening here.

I've looked into this but noticed that listening to the `abortRequested()` in the while won't even work. When a shutdown is requested, the `while not monitor.abortRequested():` will not even be evaluated since the the `xbmcgui.Dialog().yesno()` will return False and the addon will end anyway.

Here is the same code with a bit more comments, I hope this makes it more clear.
```py
    while True:
        if not kodiutils.has_credentials():
            if not kodiutils.yesno_dialog(message=kodiutils.localize(30701)):  # You need to configure your credentials...
                # The user has pressed Esc, back doesn't want to fill in credentials, or Kodi is terminating
                # Stop the Add-on
                
                kodiutils.end_of_directory()
                kodiutils.execute_builtin('ActivateWindow(Home)')
                return
           
            # The user selected Yes and wants to enter his credentials
            kodiutils.open_settings()
        else:
            # Great, we have valid credentials and want to continue outside the loop
            break

    # We have valid credentials and can continue
    
    # more code...
```